x-pack/plugins/cases/server/client/utils.ts

```diff
@@ -621,21 +623,41 @@ export const removeCustomFieldFromTemplates = ({
   }
 
   return templates.map((template) => {
-    if (!template.caseFields?.customFields || !template.caseFields?.customFields.length) {
-      return template;
-    }
+    const templateCustomFields = template.caseFields?.customFields ?? [];
 
-    if (!customFields || !customFields?.length) {
+    if (!customFields || !customFields.length) {
       return { ...template, caseFields: { ...template.caseFields, customFields: [] } };
     }
 
-    const templateCustomFields = template.caseFields.customFields.filter((templateCustomField) =>
+    // remove deleted custom field from template
+    const transformedTemplateCustomFields = templateCustomFields.filter((templateCustomField) =>
       customFields?.find((customField) => customField.key === templateCustomField.key)
     );
 
+    // add new custom fields to template
+    if (customFields.length > transformedTemplateCustomFields.length) {
```

Comment:
I am wondering, if is it possible for `customFields` and `transformedTemplateCustomFields` to be of the same size but have different custom fields. Wdyt?

Reply:
Tried to reproduce the scenario and since we remove unmatched custom fields from template before this check, the template custom field length is always less. 
however there is no harm in checking for same size as well. 😄 
Will update the condition to `if (customFields.length >= transformedTemplateCustomFields.length)`

Reply:
updated [here](https://github.com/elastic/kibana/pull/187591/commits/77f756a09a9089c29581a8cd3fa91c1a078c74c7#diff-b1977b461e735d1be9f7dad37236370734e9a0750bee631a55dbf2032ec068fa)